dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerOwnershipManager.java

```diff
@@ -104,26 +113,28 @@ public DefaultTrackerOwnershipManager(
   /** Cache for storing recent temporary ownership checks */
   private final Cache<Boolean> tempOwnerCache;
 
-  // -------------------------------------------------------------------------
-  // Implementation
-  // -------------------------------------------------------------------------
-
   @Override
   @Transactional
-  public void transferOwnership(
-      TrackedEntity trackedEntity, Program program, OrganisationUnit orgUnit)
-      throws ForbiddenException {
-    if (trackedEntity == null || program == null || orgUnit == null) {
-      return;
+  // TODO(tracker) This method should accept a tracked entity UID instead. The problem is, we can't
```

Comment:
We are solving the same problem in 2 quite different ways.
In this method we are passing the trackedEntity from the service and we know that it is already completely validated, then we need to get the managed entity to be able to save a `ProgramOwnershipHistory`. I agree that we should get only the UID and do everything in the method.

In `grantTemporaryOwnership` we are passing the UID, and we are manually validating the tracked entity (tracked entity type, existence, etc), it would be better to use `TrackedEntityService` here as well in the controller and then just retrieve the managed entity from the `manager`, so we know all the proper validations are run.